src/test/java/org/apache/commons/lang3/concurrent/MemoizerTest.java

```diff
@@ -0,0 +1,112 @@
+package org.apache.commons.lang3.concurrent;
+
+import org.easymock.EasyMockRunner;
+import org.easymock.Mock;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.easymock.EasyMock.expect;
+import static org.easymock.EasyMock.replay;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+@RunWith(EasyMockRunner.class)
+public class MemoizerTest {
```

Comment:
Wouldn't it make sense to create a test that uses an executor service to let several threads in parallel access the same Memoizer instance?